### DISCUSSION TITLE
Cypress type colon in div element with contenteditable="true"

### DISCUSSION THREAD

**Discussion by gowthamieb** [cypress-io/cypress#26021] [Questions and Help]
How to type colon in a div element whose contenteditable="true"

I have tried the below and it doesnt work. The same works as expected for input type and not for div even though contenteditable is set to true. 

cy.get('div .test')
  .type('100::20', {
    parseSpecialCharSequences: false,
  })

**Comment by portyu9** [comment]
Hi @gowthamieb 

A colon (`:`) does not need `parseSpecialCharSequences: false`. Cypress's special `.type()` sequences use braces such as `{enter}`.

The important part is to make sure Cypress is typing into the element that actually has `contenteditable="true"`.

If the element is:

```html
<div class="test" contenteditable="true">
```

use:

```js
cy.get('div.test[contenteditable="true"]')
  .click()
  .type('100::20')
  .should('have.text', '100::20')
```

Notice the selector is `div.test`, **without a space**.

`div .test` means “an element with class `test` that is a descendant of a `div`”, which can result in Cypress targeting a child rather than the contenteditable element itself.

If `.test` really is a child of the editable element, target its editable ancestor instead:

```js
cy.get('.test')
  .closest('[contenteditable="true"]')
  .click()
  .type('100::20')
```

So this should not require any special handling for `:`; first make sure `.type()` is being issued against the actual contenteditable element.

Please mark this answer as accepted if it helped, thank you.
